http/basic/src/main/java/io/cloudevents/http/HttpMessageFactory.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudevents.http;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpsExchange;
+import io.cloudevents.core.message.MessageReader;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.core.message.impl.GenericStructuredMessageReader;
+import io.cloudevents.core.message.impl.MessageUtils;
+import io.cloudevents.core.message.impl.UnknownEncodingMessageReader;
+import io.cloudevents.http.impl.CloudEventsHeaders;
+import io.cloudevents.http.impl.HttpMessageReader;
+import io.cloudevents.http.impl.HttpMessageWriter;
+
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import static io.cloudevents.http.impl.CloudEventsHeaders.CONTENT_TYPE;
+
+public class HttpMessageFactory {
+
+    public static MessageReader createReader(Consumer<BiConsumer<String,String>> forEachHeader, byte[] body) {
```

Comment:
I am thinking of adding overload for `InputStream` body since many API use that.

Reply:
```java
public static MessageReader createReader(Consumer<BiConsumer<String,String>> forEachHeader, InputStream body);
```

Reply:
I don't think that's worthwhile. Users can trivially convert an `InputStream` into a `byte[]` with [this method](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html#readAllBytes()).

Reply:
The method is not in java8, is it? Another option would be call IOUtils from apache commons, but that is yet another dependency.

Reply:
On the other hand `byte[]` is trivially convertible to `InputStream` by `ByteArrayInputStream` which is in java since `1.0`.

Reply:
Java 8 is End Of Life. I don't think new APIs should be making design decisions based on what it does or does not have. Anyway, even users who need to stay on Java 8 will not find it hard to make a `byte[]` if they have an `InputStream`.

On the other hand, if the API had _only_ `InputStream` rather than `byte[]`, I think that would be fine. As you point out, users can convert from the latter to the former trivially. And in the typical context of an HTTP server (servlet, or whatever) you typically do get the HTTP request body as an `InputStream`.

Reply:
> And in the typical context of an HTTP server (servlet, or whatever) you typically do get the HTTP request body as an InputStream.

That's why I am thinking of it. I tried to create 3 examples, all use InputStream.

Reply:
I want to keep everything consistent (as much as we can) and use a single type for the bytes. For now we use `byte[]` and i think here you should have only the `byte[]` signature, if users need it, we add the overload later for `InputStream`. But please keep everything `byte[]` based atm.
I've opened another issue for considering switchin byte[] to ByteBuffer all over the sdk https://github.com/cloudevents/sdk-java/issues/207